### PR TITLE
New Page: aria-haspopup

### PR DESCRIPTION
**Pull request by estelle**
- [X] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by estelle** [comment]
my local branch on this is borked, so hopefully this is mergeable ;)

File: files/en-us/web/accessibility/aria/attributes/aria-haspopup/index.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  - Reference
+---
+
+The `aria-haspopup` attribute indicates the availability and type of interactive popup element that can be triggered by the element on which the attribute is set.
+
+## Description
+
+In ARIA, interactive menus, listboxes, trees, grids, and dialogs that appear on top of other content when triggered to appear are considered "popups". These popups are triggered by one or more interactive elements on the page. The availability and type of popup the interactive element will trigger should be identified with the `aria-haspopup` state.
+ 
+
```
Comment (ericwbailey) [inline files/en-us/web/accessibility/aria/attributes/aria-haspopup/index.md:19]:
```suggestion
In ARIA, interactive menus, listboxes, trees, grids, and dialogs that appear on top of other content when triggered to appear are considered "popups". These popups are triggered by one or more interactive elements on the page. The availability and type of popup the interactive element will trigger should be identified with the `aria-haspopup` state.
 
```

File: files/en-us/web/accessibility/aria/attributes/aria-haspopup/index.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+The presence of the `aria-haspopup`, with one of six enumerated values - `menu`, `listbox`, `tree`, `grid`, `dialog`, or `true` - indicates the element can trigger a popup and what kind of popup will be displayed. In turn, the element that pops up must be of the role indicated. The value `true` is the same as `menu`. Any other value, including an empty string or other [role](/en-US/docs/Web/Accessibility/ARIA/Roles), is treated as if `false` were set. 
+
+A [`tooltip`](/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role), being not interactive, is not considered to be a popup in this context.
```
Comment (ericwbailey) [inline files/en-us/web/accessibility/aria/attributes/aria-haspopup/index.md:22]:
```suggestion
A [`tooltip`](/en-US/docs/Web/Accessibility/ARIA/Roles/tooltip_role) is not considered to be a popup in this context, as is not interactive.
```

File: files/en-us/web/accessibility/aria/attributes/aria-haspopup/index.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+The `aria-haspopup` state informs assistive technology users that there is a popup and the type of popup it is, but provides no interactivity. For the popup to be keyboard accessible, make sure the element with `aria-haspopup` is focusable and can trigger the popup, that there is a keyboard mechanism for opening the popup, and that the popup element manages focus of all its descendants.
+
+> **Note:** ARIA does not enable accessible functionality. ARIA only conveys the intended behavior of your functionality.
+
+When creating a [`menubar`](/en-US/docs/Web/Accessibility/ARIA/Roles/Menubar_role), a parent [`menuitem`](/en-US/docs/Web/Accessibility/ARIA/Roles/Menuitem_role) should have `aria-haspopup="menu"` (or `true`) set. Any button that open a menu should have a role of [`button`](/en-US/docs/Web/Accessibility/ARIA/Roles/Button_role) or, preferably, be a {{HTMLElement('button')}}, and also have `aria-haspopup="menu"` (or `true`) set. [`Tab`](/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_role) elements with popup menus should also have `aria-haspopup="menu"` set.
```
Comment (ericwbailey) [inline files/en-us/web/accessibility/aria/attributes/aria-haspopup/index.md:30]:
```suggestion
When creating a [`menubar`](/en-US/docs/Web/Accessibility/ARIA/Roles/Menubar_role), a parent [`menuitem`](/en-US/docs/Web/Accessibility/ARIA/Roles/Menuitem_role) should have `aria-haspopup="menu"` (or `true`) set. Any button that open a menu should have a role of [`button`](/en-US/docs/Web/Accessibility/ARIA/Roles/Button_role) or, preferably, be a {{HTMLElement('button')}}, and also have `aria-haspopup="menu"` (or `true`) set. [`Tab`](/en-US/docs/Web/Accessibility/ARIA/Roles/Tab_role) elements with popup menus should also have `aria-haspopup="menu"` set. Note that `menubar`s should not be used to create website navigation.
```

**Review by ericwbailey — CHANGES_REQUESTED** [review]

**Review by ericwbailey — APPROVED** [review]

**Pull request merged**